rflx/model.py

```diff
@@ -713,68 +718,121 @@ def __check_attributes(self, expression: Expr, location: Location = None) -> Non
-                    if isinstance(othertype, Opaque):
-                        first = Number(0)
-                        last = Number(255)
-                    elif isinstance(othertype, Array):
-                        if not isinstance(othertype.element_type, Integer):
-                            self.error.append(
-                                f'invalid array element type "{othertype.element_type.identifier}"'
-                                " for aggregate comparison",
-                                Subsystem.MODEL,
-                                Severity.ERROR,
-                                r.location,
-                            )
-                            continue
-                        first = othertype.element_type.first.simplified()
-                        last = othertype.element_type.last.simplified()
+                    return
+                first = composite.element_type.first.simplified()
+                last = composite.element_type.last.simplified()
 
-                    for element in aggregate.elements:
-                        if not first <= element <= last:
-                            self.error.append(
-                                f"aggregate element out of range {first} .. {last}",
-                                Subsystem.MODEL,
-                                Severity.ERROR,
-                                element.location,
-                            )
+            for element in aggregate.elements:
+                if not first <= element <= last:
+                    self.error.append(
+                        f"aggregate element out of range {first} .. {last}",
+                        Subsystem.MODEL,
+                        Severity.ERROR,
+                        element.location,
+                    )
+
+        def check_enumeration(relation: Relation, left: Enumeration, right: Enumeration) -> None:
+            if left != right:
+                self.error.append(
+                    "comparison of incompatible enumeration literals",
+                    Subsystem.MODEL,
+                    Severity.ERROR,
+                    relation.location,
+                )
+                self.error.append(
+                    f'of type "{left.identifier}"', Subsystem.MODEL, Severity.INFO, left.location,
+                )
+                self.error.append(
+                    f'and type "{right.identifier}"',
+                    Subsystem.MODEL,
+                    Severity.INFO,
+                    right.location,
+                )
+
+        def relation_error(relation: Relation, left: TypeExpr, right: TypeExpr) -> None:
+            self.error.append(
+                f'invalid relation "{relation.symbol}" between {left.__class__.__name__} '
+                f"and {right.__class__.__name__}",
+                Subsystem.MODEL,
+                Severity.ERROR,
+                relation.location,
+            )
+
+        def __resolve_types(
```

Comment:
Inconsistent naming: `__resolve_types` and `__invalid_relation` are the only local functions prefixed with `__`. I see no need for having this prefix on local functions.

Reply:
Fixed in 4fad579.